serverless/resource_serverless.go

```diff
@@ -0,0 +1,158 @@
+package serverless
+
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceServerless() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceServerlessCreate,
+		Read:   resourceServerlessRead,
+		Update: resourceServerlessUpdate,
+		Delete: resourceServerlessDelete,
+
+		Schema: map[string]*schema.Schema{
+			"config_dir": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"serverless_bin_dir": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+			},
+			// The directory where the Serverless package lives. In the CLI, this defaults to
+			// .serverless, but we default to .terraform-serverless to avoid an issue where
+			// the CLI deletes the .serverless directory after deploy, even with --package.
+			// Note that the provider requires out-of-band packaging. Users should package
+			// their code with `sls package --package .terraform-serverless`.
+			//
+			// NOTE: the path you provide must be RELATIVE to your `config_dir` since the
+			// --package flag in the CLI does not support absolute paths.
+			"package_dir": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  ".terraform-serverless",
```

Comment:
Can we take this information and get it into the appropriate place in the README? The vast bulk of folks familiar with serverless would assume I think it should default to `.serverless`, so we should explain why the default behavior won't work and you should choose our default or something that is _not_ `.serverless`

Reply:
Captured work in https://github.com/FormidableLabs/terraform-provider-serverless/issues/9